scio-avro/src/main/scala/com/spotify/scio/avro/syntax/ScioContextSyntax.scala

```diff
@@ -49,6 +50,14 @@ final class ScioContextOps(private val self: ScioContext) extends AnyVal {
   def avroFile[T: ClassTag: Coder](path: String, schema: Schema): SCollection[T] =
     self.read(GenericRecordIO[T](path, schema))
 
+  /**
+   * Get an SCollection of type [[T]] for data stored in Avro format after applying
+   * parseFn to map a serialized [[org.apache.avro.generic.GenericRecord GenericRecord]]
+   * to type [[T]]
+   */
+  def avroFile[T: Coder](path: String, parseFn: GenericRecord => T): SCollection[T] =
```

Comment:
🎨 can we curry this?

Reply:
I can try this.

Reply:
I tried a few things, but I am unable to figure out the correct syntax.
 - `def avroFile[T: Coder](path: String)(parseFn: GenericRecord => T)`
This one is not working because reading `SpecificRecord` is by using `def avroFile[T: <: SpecificRecord: Coder](path: String)` and hence the first function fails because the compiler wont be able to resolve the overloaded method.
 - Shall we try and rename this api? something like `parsedAvroFile` ?
 - ~The other option is to add "not a sub type" implicit~
```
import shapeless.<:!<
def avroFile[T](path: String)(parseFn: GenericRecord => T)(
    implicit toder: Coder[T],
    notSpecificRecord: T <:!< SpecificRecord
  )
```
or
```
import shapeless._
def avroFile[T: |¬|[SpecificRecord]#λ: Coder](path: String)(parseFn: GenericRecord => T)
```
This actually brings in an extra requirement that `T` should not be a SpecificRecord, which can be a little too restricting, considering that the api doesn't need to impose that. But still this doesn't compile with currying.

Reply:
Had a brief look and yes we do have ambiguous resolution here when we curry. I’m leaning towards renaming it to your suggestion or just `parseAvroFile`. I think this gives a better syntax when expressing the parsing function